### PR TITLE
fix(vdom): Hide fallback slot when content is entered

### PR DESCRIPTION
**Pull request by kevin-coyle-unipro**
Since version 1.8.7 slotted content shows both the fallback content and the inserted content. When shadow is unavailable and it is using scoped slots.

Originally the tests passed because there didn't seem to be a test case on this scenario. In this PR we have added an additional test case to check the attribute of the slot and removed the conditional that prevented the hidden attribute from being added.

This seems to relate to issue https://github.com/ionic-team/stencil/issues/2307 which although was closed does not seem to be fixed.

### CODE REVIEW

**Comment by kevin-coyle-unipro** [comment]
Hey I'd like to see if it's possible that we can get this in. Unfortunately without some kinda of fix for this both the fallback slot and the intended slots are visible in Legacy Edge and IE11.

**Comment by viljamis** [comment]
Hmm we’re seeing this issue in all browsers; latest Chrome, Safari and Firefox too. default slot content is always shown, no matter what.

File: src/runtime/vdom/test/scoped-slot.spec.tsx [hunk 1]
```diff
@@ -755,4 +755,26 @@ describe('scoped slot', () => {
     // expect(root.firstElementChild.firstElementChild.children[1].firstElementChild.children[1].nodeName).toBe('GOAT');
     // expect(root.firstElementChild.firstElementChild.children[1].firstElementChild.children[1].textContent).toBe('hey goat!');
   });
+
+  it('should hide the slot\'s fallback content when slot content passed in', async() => {
+    @Component({ tag: 'fallback-test', scoped: true })
```
Comment (simonhaenisch) [inline src/runtime/vdom/test/scoped-slot.spec.tsx:760]: So this happens when setting `scoped: true`? I think it would also happen when adding `shadow: false`. Could you also add a test for that?

File: src/runtime/vdom/vdom-render.ts [hunk 2]
```diff
@@ -418,24 +418,22 @@ const updateFallbackSlotVisibility = (elm: d.RenderNode) => {
...
-              }
-            } else {
-              // this is a default fallback slot node
-              // any element or text node (with content)
-              // should hide the default fallback slot node
-              if (nodeType === NODE_TYPE.ElementNode || (nodeType === NODE_TYPE.TextNode && childNodes[j].textContent.trim() !== '')) {
-                childNode.hidden = true;
-                break;
-              }
+          // this sibling node is from a different component
```
Comment (simonhaenisch) [inline src/runtime/vdom/vdom-render.ts:421]: This comment was regarding the `if (childNodes[j]['s-hn'] !== childNode['s-hn'])` statement, so should also be deleted. Wondering whether this condition was important for some other edge cases though (which might also not have a test yet).

Maybe sth like

```ts
if (childNodes[j]['s-hn'] !== childNode['s-hn'] || slotNameAttr !== '') {
  // ...
}
```

would be better?

**Comment by viljamis** [comment]
@kevin-coyle-unipro Do you think you’ll have time to work further with this or would you want some help here? Happy to help if needed.

**Comment by kevin-coyle-unipro** [comment]
> @kevin-coyle-unipro Do you think you’ll have time to work further with this or would you want some help here? Happy to help if needed.

Hey! I'd love some help if its on offer :)

**Comment by viljamis** [comment]
@kevin-coyle-unipro Would you be willing to share access to your fork, or should I create my own?

**Comment by viljamis** [comment]
@simonhaenisch @kevin-coyle-unipro We ended up creating a new pull for additional fixes as that was easier. Please see https://github.com/ionic-team/stencil/pull/2650

**Comment by claviska** [comment]
Thanks for the PR. We're going to focus on getting it merged via #2650 so please subscribe to that issue for updates.
